google-cloud-trace/lib/google/cloud/trace/patches/call_with_trace.rb

```diff
@@ -0,0 +1,87 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/trace"
+
+module GRPC
+  module Core
+    ##
+    # Instrumentation Stackdriver Trace in GRPC by patching GRPC::Core::Call
+    # class. Intercept each GRPC request and create a Trace span with basic
+    # request information.
+    module CallWithTrace
+      ##
+      # @private Add request labels from the Call object and message.
+      def self.add_request_labels call, message, labels
+        label_keys = Google::Cloud::Trace::LabelKey
+
+        message_size = message.to_s.bytesize.to_s
+
+        set_label labels, label_keys::RPC_REQUEST_SIZE, message_size
+        set_label labels, label_keys::RPC_HOST, call.peer.to_s
```

Comment:
Is there not a way to get the function name being called? e.g. `entries:write` for logging? Or is that encoded in the host?